src/lib/mdns/ServicePool.h

```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/mdns/platform/Mdns.h>
+
+namespace chip {
+namespace Mdns {
+
+constexpr uint64_t kUndefinedNodeId = 0;
+
+class ServicePool
```

Comment:
Needs documentation and unit tests.

I found hard to follow what the class did internally. Especially the 'remove random' bits.

Reply:
The unit test is in src/lib/mdns/tests/TestServicePool.cpp

Reply:
Please add documentation. Add/remove/update/clear are obvious.

ReHash and ShouldReHash are not clear to me.